bindings/pyroot/pythonizations/python/ROOT/_pythonization/_function_utils.py

```diff
@@ -0,0 +1,224 @@
+        """
+        try:
+            import inspect 
+        except:
+            raise ImportError("Failed to import inspect during call to determine function signature.")
+        func_sign = inspect.signature(func)
+        # Find the Return type
+        if func_sign.return_annotation is inspect._empty:
+            raise ValueError("Return type of the function is not mentioned.\n Function cannot be jitted as signature cannot be determined.")
+        else:
+            self.return_type = str(func_sign.return_annotation)
+        
+        # List of input parameters for function
+        self.params = list(func_sign.parameters.keys()) # ALl the input parameters the function needs.
+        input_args = {}
+        type_args = {}
+
+        # if len(funcargs) != 0:
+        for p in self.params:
+            if p in funcargs.keys(): # the parameter value has been given in funcargs
+                p_value = funcargs[p]
+                type_args[p] = self.find_type(p_value)
+                if type(p_value) == bool: # Bools in python are represented as True/False but in C++ are true/false. The followinf if is to account for that
+                    if p_value: p_value = 'true'
+                    else: p_value = 'false'
+                input_args[p] = p_value
+            else: # the parameter was not in funcargs. Thus this parameter has to be mapped to a column of rdf
+                if p not in self.col_names:
+                    raise Exception(f"Unable to map function argument {p} to a column.\nUse correct name of column or use funcargs.")
+                input_args[p] = p 
+                type_args[p] = self.find_type(p)
+        return input_args, type_args
+
+
+    def generate_function_call(self, func, funcargs):
+        """
+        Function to generate the c++ function call from the Numba Namespace.
+        Also gives a name to the function if the function is a lambda function.
+        Args:
+	    `   func: Function whose signature is to be determined.
+                    P.S The function should not be crazy. It should have simple datatypes for the arguments and should not have default arguments. (Use funcargs instead)
+            funcargs: Dictionatry containing constant parameters the that func requires. Or any special mapping of the arguments of func.
+
+        Returns:
+            func_call: String of function's c++ call which can be passed to the corresponding define or filter.
+            func_sign: The signature of the function which is needed for jitting.
+        """
+        input_args, type_args = self.find_function_signature(func, funcargs)
+        if func.__name__ == '<lambda>' : 
+            try:
+                from uuid import uuid4
+            except:
+                raise ImportError("Could not import uuid module to give unique identifier to lambda functions.")
+            func.__name__ = "func_"+uuid4().__str__().split('-')[-1]
```

Comment:
Can we perhaps change this random ID into a sequential ID using a class-level counter?